torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -5009,30 +5013,36 @@ def test_DistributedDataParallel_SyncBatchNorm_Channels_Last(self):
             global_bs = int(num_processes * 2)
 
             model = ONLY_SBN_NET
-            model_gpu = copy.deepcopy(model).cuda(rank)
+            if torch.cuda.is_available():
+                device = torch.device(f"cuda:{rank}")
+                device_ids = [rank]
+            else:
+                device = torch.device('cpu')
+                device_ids = None
+            model_base = copy.deepcopy(model).to(device)
             model_DDP = nn.parallel.DistributedDataParallel(
-                model_gpu, device_ids=[rank]
+                model_base, device_ids=device_ids
             )
 
             memory_format = torch.channels_last
-            input_gpu = (
+            input_base = (
                 torch.randn(global_bs, 2, 4, 4, dtype=torch.float)
-                .cuda(rank)
+                .to(device)
```

Comment:
i wonder if to(device), to(memory_format) could be integrated directly as arguments to randn(...). It seems that device = device definitely could be moved there, but memory_format seems not supported by randn(...)

Reply:
Indeed, `to(device)` could be integrated in `randn`, but `memory_format` could not.

`@overload`
`def randn(size: _size, *, generator: Optional[Generator], names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(*size: _int, generator: Optional[Generator], names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(size: _size, *, generator: Optional[Generator], out: Optional[Tensor]=None, dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(*size: _int, generator: Optional[Generator], out: Optional[Tensor]=None, dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(size: _size, *, out: Optional[Tensor]=None, dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(*size: _int, out: Optional[Tensor]=None, dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(size: _size, *, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`